### DISCUSSION TITLE
libgomp is not static linked in manylinux2014 cpu binary

### DISCUSSION THREAD

**Discussion by jurassic410** [TabbyML/tabby#2886] [Q&A] [answered]
**Describe the bug**
hello, i don't know id it's a bug or i missed something, but on a clean installation of ubuntu 24.04, i get an error with shared libraries, i tried to install locally downloading the zip as decribed in the documentation, but i get:
`WARN llama_cpp_server::supervisor: crates/llama-cpp-server/src/supervisor.rs:108: <embedding>: /home/administrator/dist/tabby_x86_64-manylinux2014/llama-server: error while loading shared libraries: libgomp.so.1: cannot open shared object file: No such file or directory`
so i tried to go down the docker route, but a similar problem appear but with a different shared library:
`WARN llama_cpp_server::supervisor: crates/llama-cpp-server/src/supervisor.rs:111: <embedding>: /opt/tabby/bin/llama-server: error while loading shared libraries: libcuda.so.1: cannot open shared object file: No such file or directory
` i found another issue opened that suggest to use v0.13.1 but it seems that the same problem is present

**Information about your version**
v0.15.0

**Information about your GPU**
CPU Only

**Comment by wsxiaoys** [comment]
Seems like a bug to me - libgomp1 shall be linked statically

**Comment by jurassic410** [comment]
my usual luck :-(
there is something that i can try to do to help?

**Comment by wsxiaoys** [comment]
After a second look, it does seem that libgomp1 has to be linked dynamically. Could you try running `apt install libgomp1` in your Linux environment and then run the CPU manylinux2014 binary again?

**Comment by wsxiaoys** [comment]
adding https://github.com/TabbyML/tabby/pull/2866 as a potential fix

**Comment by jurassic410** [comment, marked as answer]
> After a second look, it does seem that libgomp1 has to be linked dynamically. Could you try running `apt install libgomp1` in your Linux environment and then run the CPU manylinux2014 binary again?

can confirm that installing libgomp1 manually works!

**Comment by wsxiaoys** [comment]
Great! Given that OpenMP is such a foundational dependency, I would prefer to keep it dynamically linked for now.
